packages/test-utils/src/testUtils/apis/ErrorApi/MockErrorApi.test.ts

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { MockErrorApi } from './MockErrorApi';
+
+async function ifResolved<T>(promise: Promise<T>): Promise<T | 'not-yet'> {
+  return Promise.race([promise, Promise.resolve<'not-yet'>('not-yet')]);
```

Comment:
Wonder if there's a better way to model this rather than `not-yet` maybe using `Promise.resolve(false)` or `Promise.reject`? As the curious behaviour of `Promise.race` suggests that even if you fall over flat on your face half way through the race, you still finished first and win the race.

Reply:
I mean it's fine for now, just a smell.